File: AVS/task2/src/animals/fish/Fish.cpp
//
// Created by Матвей Попов on 14.10.2021.
//

#include "Fish.h"

Fish::Fish(const std::string &name, const int32_t &weight): Animal(name, weight) {  }

Fish::Fish(const Fish &fish): Animal(fish) {
    location = fish.location;
}

Fish::~Fish() noexcept {

}

Fish& Fish::operator=(const Fish &fish) {
    Animal::operator=(fish);
    location = fish.location;
    return *this;
}

Animal * Fish::copy() {
    Fish *currentFish = dynamic_cast<Fish *>(this);
    Fish *copyFish = new Fish(*name, weight);
    copyFish->setLocation(currentFish->getLocation());
    return copyFish;
}

Fish::Location Fish::getLocation() const {
    return location;
}

void Fish::setLocation(const Location &location) {
    this->location = location;
}

File: AVS/task3/src/animals/fish/Fish.py
from src.animals.animal.Animal import Animal
import enum


class Fish(Animal):

    def __init__(self, location, name, weight):
        super(Fish, self).__init__(name, weight)
        self.location = location
        if location is not Location:
            print("Invalid location!\n")
            print("Setting to default: " + Location.sea.name + "\n")
            self.location = Location.sea

    def get_location(self):
        return self.location

    def get_info_data(self):
        return "This is fish\n" + \
               "The location s " + self.location.name + "\n" + \
               super(Fish, self).get_info_data() + "\n\n"


class Location(enum.Enum):
    sea = "Sea"
    ocean = "Ocean"
    lake = "Lake"


def get_location_by_name(name):
    if name == "Sea":
        return Location.sea
    elif name == "Ocean":
        return Location.ocean
    elif name == "Lake":
        return Location.lake
    else:
        return Location.sea
